kolibri/core/templatetags/kolibri_tags.py

```diff
@@ -40,3 +45,46 @@ def kolibri_main_navigation():
             "window._nav={0};"
             "</script>".format(json.dumps(init_data)))
     return mark_safe(html)
+
+@register.simple_tag(takes_context=True)
+def kolibri_bootstrap_model(context, base_name, api_resource, **kwargs):
+    response, kwargs = _kolibri_bootstrap_helper(context, base_name, api_resource, 'detail', **kwargs)
+    html = ("<script type='text/javascript'>"
+            "var model = {0}.resources.{1}.createModel(JSON.parse({2}));"
+            "model.synced = true;"
+            "</script>".format(settings.KOLIBRI_CORE_JS_NAME, api_resource, JSONRenderer().render(response.content)))
+    return mark_safe(html)
+
+@register.simple_tag(takes_context=True)
+def kolibri_bootstrap_collection(context, base_name, api_resource, **kwargs):
+    response, kwargs = _kolibri_bootstrap_helper(context, base_name, api_resource, 'list', **kwargs)
+    html = ("<script type='text/javascript'>"
+            "var collection = {0}.resources.{1}.createCollection({2}, JSON.parse({3}));"
+            "collection.synced = true;"
+            "</script>".format(settings.KOLIBRI_CORE_JS_NAME, api_resource, json.dumps(kwargs), JSONRenderer().render(response.content)))
+    return mark_safe(html)
+
+def _replace_dict_values(check, replace, dict):
+    for (key, value) in dict.iteritems():
+        if dict[key] is check:
+            dict[key] = replace
+
+def _kolibri_bootstrap_helper(context, base_name, api_resource, route, **kwargs):
+    # instantiate client instance to make requests to server
+    client = APIClient()
+    # copy session key to client to make requests on behalf of user
+    client.cookies[settings.SESSION_COOKIE_NAME] = context['request'].session.session_key
+    reversal = dict()
+    kwargs_check = 'kwargs_'
+    # remove prepended string and matching items from kwargs
+    for key in kwargs.keys():
+        if kwargs_check in key:
+            item = kwargs.pop(key)
+            key = re.sub(kwargs_check, '', key)
+            reversal[key] = item
+    url = reverse('{0}-{1}'.format(base_name, route), kwargs=reversal)
+    # switch out None temporarily because invalid filtering and caching can occur
+    _replace_dict_values(None, str(''), kwargs)
+    response = client.get(url, data=kwargs)
```

Comment:
It seems a bit "dirty" to be using the [test HTTP client](https://docs.djangoproject.com/en/1.10/topics/testing/tools/#the-test-client) here - I don't think it was designed for this purpose.

I believe it should be possible to get a reference to the [target renderer](http://www.django-rest-framework.org/api-guide/renderers/) and [call it programmatically](http://stackoverflow.com/questions/26790981/how-to-programatically-call-a-django-rest-framework-view-within-another-view).